src/workloads/execution/ExpressiveQueries.yml

```diff
@@ -0,0 +1,207 @@
+
+- Name: InsertData
+  Type: Loader
+  Threads: 1 # Loader doesn't support multiple threads for a single collection
+  Phases:
+  - Repeat: 1
+    Database: &DB test
+    Threads: 1
+    CollectionCount: 1 # Collection name will be Collection0, this is not configurable.
+    DocumentCount: 1000000 # 1M
+    BatchSize: 1000
+    Document:
+      region: {^RandomInt: {min: 0, max: 1999}}
+      firstname: {^RandomString: {length: {^RandomInt: {min: 3, max: 10}}}}
+      lastname: {^RandomString: {length: {^RandomInt: {min: 5, max: 10}}}}
+      cell: {^RandomInt: {min: 1111111111, max: 9999999999}}
+      email: {^RandomString: {length: {^RandomInt: {min: 10, max: 30}}}}
+      dob: {^RandomDate: {min: "1930-01-01", max: "2016-12-31"}}
+      gender: {^Choose: {from: ["Male", "Female", "Other"]}}
+      address: {
+        number: {^RandomInt: {min: 1, max: 9999}},
+        street: {^RandomString: {length: {^RandomInt: {min: 30, max: 60}}}},
+        city: {^RandomString: {length: {^RandomInt: {min: 5, max: 10}}}},
+        state: {^Choose: {from: ["AL", "AK", "AZ", "AR", "CA", "CO", "CT", "DE", "DC", "FL", "GA", "HI", "ID", "IL", "IN", "IA", "KS", "KY", "LA", "ME", "MD", "MA", "MI", "MN", "MS", "MO", "MT", "NE", "NV", "NH", "NJ", "NM", "NY", "NC", "ND", "OH", "OK", "OR", "PA", "RI", "SC", "SD", "TN", "TX", "UT", "VT", "VA", "WA", "WV", "WI", "WY"]}},
+        zip: {^RandomString: {length: 6}},
+        location: {position: {type: "Point", coordinates: [
+          {^RandomDouble: {min: -118.668469, max: -82.062023}},
+          {^RandomDouble: {min: 32.924436, max: 41.679986}}
+        ]}}
+      }
+      policies: {^Array: {number: {^RandomInt: {min: 3, max: 6}}, of: {^Choose: {from: [
+        {
+          policyType: "auto",
+          policyNum: {^RandomString: {length: 30}},
+          nextRenewalDt: {^RandomDate: {min: "2017-01-01", max: "2017-12-31"}},
+          year: {^RandomInt: {min: 1980, max: 2017}},
+          value: {^RandomInt: {min: 500, max: 500000}},
+          model: {^RandomString: {length: 30}},
+        },
+        {
+          policyType: "home",
+          policyNum: {^RandomString: {length: 30}},
+          nextRenewalDt: {^RandomDate: {min: "2017-01-01", max: "2017-12-31"}},
+          year: {^RandomInt: {min: 1970, max: 2017}},
+          value: {^RandomInt: {min: 500, max: 10000000}},
+          address: {
+            number: {^RandomInt: {min: 1, max: 9999}},
+            street: {^RandomString: {length: {^RandomInt: {min: 30, max: 60}}}},
+            city: {^RandomString: {length: {^RandomInt: {min: 5, max: 10}}}},
+            state: {^Choose: {from: ["AL", "AK", "AZ", "AR", "CA", "CO", "CT", "DE", "DC", "FL", "GA", "HI", "ID", "IL", "IN", "IA", "KS", "KY", "LA", "ME", "MD", "MA", "MI", "MN", "MS", "MO", "MT", "NE", "NV", "NH", "NJ", "NM", "NY", "NC", "ND", "OH", "OK", "OR", "PA", "RI", "SC", "SD", "TN", "TX", "UT", "VT", "VA", "WA", "WV", "WI", "WY"]}},
+            zip: {^RandomString: {length: 6}},
```

Comment:
Consider {^RandomString: {length: 6, alphabet: "0123456789"}}. I think that would be closer to what you intend.